include/gigamonkey/p2p/association_id.hpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2021 Katrina Knight
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef GIGAMONKEY_P2P_ASSOCIATION_ID_HPP_
+#define GIGAMONKEY_P2P_ASSOCIATION_ID_HPP_
+
+#include <boost/shared_ptr.hpp>
+#include <boost/uuid/uuid.hpp>
+#include "data/cross.hpp"
+#include "gigamonkey/types.hpp"
+namespace Gigamonkey::Bitcoin::P2P {
+	class AssociationID {
```

Comment:
Is there any reason for this to be an abstract type? Would we ever use any other kind of association ID?